### PR TITLE
Remove grumpy-mode-incompatible isset in payment notification

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Remove grumpy-mode-incompatible isset in payment notification

Before
----------------------------------------
issets cause fatal error in smarty grumpy mode

After
----------------------------------------
If tests pass should be ok

Technical Details
----------------------------------------
This could cause changes to output for $0 amounts (possibly) - but the template is obscure enough & the change in output minor enough that I don't think it's worth delving into

Comments
----------------------------------------
I'm not sure if this will pass yet - enotices could re-emerge but if they don't this should be mergeable

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Hmm test fail is because 0 balance not shown now - I'm rather inclined to fix the test to the new behaviour - Balance Owed seems a bit redundant here - any opinion @seamuslee001 @colemanw @demeritcowboy  ?

image

**Comment by seamuslee001** [comment]
I think that @KarinG would be useful here IMO

**Comment by KarinG** [comment]
That's just receipt / printing - I think that's ok.

**Comment by eileenmcnaughton** [comment]
Ok since @KarinG wasn't concerned I updated the tests to balance is not expecte if $0

File: tests/phpunit/api/v3/PaymentTest.php [hunk 1]
```diff
@@ -22,20 +22,8 @@ class api_v3_PaymentTest extends CiviUnitTestCase {
...
   protected $_financialTypeId = 1;
 
-  /**
-   * Should financials be checked after the test but before tear down.
-   *
-   * Ideally all tests (or at least all that call any financial api calls ) should do this but there
-   * are some test data issues and some real bugs currently blocking.
-   *
-   * @var bool
-   */
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/PaymentTest.php:32]: This is now the same as the parent - we opt out not into this check now

**Pull request merged**